rfcs/0009_Unified_Static_Memory_Planning.md

```diff
@@ -0,0 +1,473 @@
+       intermediate_tensor_1 = tir.allocate(..., pinned_memory=[1, 2])
+       intermediate_tensor_2 = tir.allocate(..., pinned_memory=[1, 2])
+       weights = tir.allocate_const(..., pinned_memory=["itcm", "flash"])
+       biases = tir.allocate_const(..., pinned_memory=["itcm", "flash"])
+       ...
+       <compute>
+       ...
+```
+##### Step 2 : Run an analysis pass to populate a Map<tir::Var, BufferInfo> that contains buffer information as defined above (See the struct BufferInfo).
+
+##### Step 3 : Use the updated Map<tir::Var, BufferInfo> to generate Array<BufferInfo>, Map<String, Integer> pool_sizes
+
+##### Step 4 : Call the provided/default algorithm (void (*foo)(Array<ByfferInfo> buffers, Map<String, Integer> pool_sizes) to populate pool_id and pool_offset.
+
+##### Step 5 : Use the updated Map<tir::Var, BufferInfo> (with pool_id and pool_offset) mutate the IR that would result as following :
+```
+# This snippet shows the format of post-USMP pseudo TIR code.
+
+    def main(input1: ty.handle, output1: ty.handle, params_1 : ty.handle, params_2 : ty.handle, workspace_1 : ty.handle, workspace_2 : ty.handle):
+       tir.call("my_model_fused_op1", input1, params1, params2, workspace_1, workspace_2)
+       tir.call("my_model_fused_op2", params1, params2, workspace_1, workspace_2)
+
+    def my_model_fused_op1(input, params_1, params_2, workspace_1, workspace_2):
+       tir.func_attr({"global_symbol":"my_model_fused_op1","tir.noalias":True})
+       intermediate_tensor_1=tir.load("uint8", workspace_1.data, <offset>)
+       intermediate_tensor_2=tir.load("uint8", workspace_1.data, <offset>)
+       output=tir.load("uint8", workspace_1.data, <offset>)
+       weights=tir.load("uint8", params_1.data, <offset>)
+       biases=tir.load("uint8", params_1.data, <offset>)
+       ...
+       <compute>
+       ...
+
+    def my_model_fused_op2(params_1, params_2, workspace_1, workspace_2):
+       tir.func_attr({"global_symbol":"my_model_fused_op2","tir.noalias":True})
+       input=tir.load("uint8", workspace_1.data, <offset>)
+       intermediate_tensor_1=tir.load("uint8", workspace_1.data, <offset>)
+       intermediate_tensor_2=tir.load("uint8", workspace_2.data, <offset>)
+       output=tir.load("uint8", workspace_2.data, <offset>)
+       weights=tir.load("uint8", params_1.data, <offset>)
+       biases=tir.load("uint8", params_2.data, <offset>)
+       ...
+       <compute>
+       ...
+```
+# Code Structure
+
+* src/tir/usmp/analysis/ -- this is where analysis pases of USMP will live
+* src/tir/usmp/transforms/ -- this is where transform pases of USMP will live
+* src/tir/usmp/usmp.cc -- this is main intergration of USMP that exposes the full TIR --> TIR transformation as described.
+* tests/python/unittest/test_tir_usmp_*.py -- this where unittests for each of the passes and pass pipeline for USMP as a component will live.
+
+NOTE 1: All the above passes will have a mirror in the python.
```

Comment:
could you define "mirror" here? also s/the python/python\//

Reply:
Done